File: python/746. Min Cost Climbing Stairs/solution.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        min_costs = [0] * (len(cost) + 1)

        for i in range(2, len(cost) + 1):
            c1 = min_costs[i - 1] + cost[i - 1]
            c2 = min_costs[i - 2] + cost[i - 2]
            min_costs[i] = min(c1, c2)

        return min_costs[-1]

File: java/746. Min Cost Climbing Stairs/Solution.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int[] minCosts = new int[cost.length + 1];

        minCosts[0] = 0;
        minCosts[1] = 0;

        for (int i = 2; i < minCosts.length; i++) {
            int c1 = minCosts[i - 1] + cost[i - 1];
            int c2 = minCosts[i - 2] + cost[i - 2];
            minCosts[i] = Math.min(c1, c2);
        }
        return minCosts[minCosts.length - 1];
    }
}